recipes/gdal/all/conanfile.py

```diff
@@ -834,7 +834,7 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "GDAL")
-        self.cpp_info.set_property("cmake_target_name", "GDAL")
+        self.cpp_info.set_property("cmake_target_name", "GDAL::GDAL")
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "GDAL::GDAL")
        self.cpp_info.set_property("cmake_find_mode", "both") 
```

https://cmake.org/cmake/help/latest/module/FindGDAL.html